bazel/repository_locations.bzl

```diff
@@ -171,130 +210,155 @@ REPOSITORY_LOCATIONS = dict(
         strip_prefix = "zlib-79baebe50e4d6b73ae1f8b603f0ef41300110aa3",
         # 2019-04-14 development branch
         urls = ["https://github.com/madler/zlib/archive/79baebe50e4d6b73ae1f8b603f0ef41300110aa3.tar.gz"],
+        use_category = ["dataplane"],
     ),
     com_github_jbeder_yaml_cpp = dict(
         sha256 = "77ea1b90b3718aa0c324207cb29418f5bced2354c2e483a9523d98c3460af1ed",
         strip_prefix = "yaml-cpp-yaml-cpp-0.6.3",
         urls = ["https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.6.3.tar.gz"],
+        use_category = ["dataplane"],
     ),
     com_github_msgpack_msgpack_c = dict(
         sha256 = "433cbcd741e1813db9ae4b2e192b83ac7b1d2dd7968a3e11470eacc6f4ab58d2",
         strip_prefix = "msgpack-3.2.1",
         urls = ["https://github.com/msgpack/msgpack-c/releases/download/cpp-3.2.1/msgpack-3.2.1.tar.gz"],
+        use_category = ["dataplane"],
```

Comment:
Who uses this?

Reply:
It is used by the datadog tracer. If we'll decide  have separate category for tracers I'll reassign.

Reply:
Changed `use_category` to `observability`